libsolidity/ast/Types.cpp

```diff
@@ -2773,18 +2773,9 @@ FunctionTypePointer FunctionType::asMemberFunction(bool _inLibrary, bool _bound)
 			kind = Kind::DelegateCall;
 	}
 
-	TypePointers returnParameterTypes = m_returnParameterTypes;
-	if (kind != Kind::Internal)
-	{
-		// Alter dynamic types to be non-accessible.
-		for (auto& param: returnParameterTypes)
-			if (param->isDynamicallySized())
-				param = make_shared<InaccessibleDynamicType>();
```

Comment:
We could move this check into the `TypeChecker` around line 1500.

Reply:
... only if the pragma specifies that we are using the old encoder.